packages/ts-moose-lib/src/scripts/activity.ts

```diff
@@ -104,11 +105,31 @@ export const activities = {
     task: Task<any, any>,
     inputData: any,
   ): Promise<any[]> {
+    // Get context for heartbeat (required for cancellation detection)
+    const context = Context.current();
+
+    // Periodic heartbeat. Important for cancellation detection.
+    let heartbeatInterval: NodeJS.Timeout | null = null;
```

Comment:
I don't get why you need the heartbeat, can you explain?

Reply:
Good call, I added more comments & a link to their doc. Seems to imply that the cancellation signal is part of the heartbeat response from temporal server and that tells the `context.cancelled` to reject. The reject didn't work without the heartbeat.